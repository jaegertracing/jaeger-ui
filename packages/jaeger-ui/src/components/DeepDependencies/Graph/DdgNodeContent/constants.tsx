// Copyright (c) 2019 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { getUrl as getSearchUrl } from '../../../SearchTracePage/url';

// Constants for calculating circular node positioning
export const FONT_SIZE = 14;
export const FONT = `${FONT_SIZE}px Helvetica Nueue`;
export const LINE_HEIGHT = 1.5;
export const OP_PADDING_TOP = 5;
export const PROGRESS_BAR_STROKE_WIDTH = 15;
export const RADIUS = 75;
export const WORD_RX = /\W*\w+\W*/g;

// While browsers suport URLs of unlimited length, many server clients do not handle more than this max
// export const MAX_LENGTH = 2083;
// get unpreditable, unreproducable results past 7400 characters
export const MAX_LENGTH = 7000;
export const MAX_LINKED_TRACES = 35;
export const MIN_LENGTH = getSearchUrl().length;
export const PARAM_NAME_LENGTH = '&traceID='.length;
