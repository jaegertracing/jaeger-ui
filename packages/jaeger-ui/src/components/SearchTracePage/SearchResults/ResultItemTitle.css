/*
Copyright (c) 2017 Uber Technologies, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.ResultItemTitle {
  background: #ececec;
  border-bottom: 1px solid #d8d8d8;
  display: flex;
}

.ResultItemTitle--item {
  color: inherit;
  padding: 0.5rem;
  position: relative;
}

.ResultItemTitle--item:first-child {
  border-right: 1px solid #ddd;
}

.ResultItemTitle--item:hover {
  background: #e4e4e4;
  border-color: #ccc;
}

.ResultItemTitle--title {
  margin: 0;
  position: relative;
}

.ResultItemTitle--title.is-error {
  color: #c00;
}

.ResultItemTitle--idExcerpt {
  color: #888;
  font-weight: normal;
  padding-left: 0.5rem;
}

.ResultItemTitle--durationBar {
  background: #d7e7ea;
  bottom: 0;
  left: 0;
  position: absolute;
  top: 0;
}

.ResultItemTitle--item:hover > .ResultItemTitle--durationBar {
  background: #c5dde0;
}
