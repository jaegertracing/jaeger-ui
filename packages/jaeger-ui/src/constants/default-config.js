// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

export const FALLBACK_DAG_MAX_NUM_SERVICES = 100;
export const FALLBACK_TRACE_NAME_REGEXP_ALLOWLIST = [];

export const deprecations = [
  {
    formerKey: 'dependencies.dagMaxNumServices',
    currentKey: 'dependencies.menuEnabled',
    formerValue: 100,
    currentValue: false,
    callback: (config, formerKey, currentKey, formerValue, currentValue) => {
      const dagMaxNumServices = config.dependencies && config.dependencies.dagMaxNumServices;
      if (dagMaxNumServices !== FALLBACK_DAG_MAX_NUM_SERVICES) {
        // set the current key to true because the former value was not the default
        if (config.dependencies) {
          config.dependencies.menuEnabled = true;
        } else {
          config.dependencies = { menuEnabled: true };
        }
        // eslint-disable-next-line no-console
        console.warn(
          `Deprecated configuration key (${formerKey}: ${dagMaxNumServices}) detected, ` +
            `please use (${currentKey}: true) instead.`
        );
      }
    },
  },
];

export default {
  // Setting archiveEnabled to null means:
  // 1. Respect backend capability for non-memory storage
  // 2. Disable for in-memory storage (handled in getArchivedEnabled function)
  archiveEnabled: null,
  dependencies: {
    dagMaxNumServices: FALLBACK_DAG_MAX_NUM_SERVICES,
    menuEnabled: true,
  },
  search: {
    maxLookback: {
      label: '2 Days',
      value: '2d',
    },
    maxLimit: 1500,
  },
  tracking: {
    gaID: null,
    trackErrors: true,
    customWebAnalytics: null,
  },
  linkPatterns: [],
  menu: [
    {
      label: 'About Jaeger',
      items: [
        {
          label: 'GitHub',
          url: 'https://github.com/jaegertracing/jaeger',
        },
        {
          label: 'Docs',
          url: 'https://www.jaegertracing.io/docs/latest',
        },
        {
          label: 'Twitter',
          url: 'https://twitter.com/JaegerTracing',
        },
        {
          label: 'Discussion Group',
          url: 'https://groups.google.com/forum/#!forum/jaeger-tracing',
        },
        {
          label: 'Gitter.im',
          url: 'https://gitter.im/jaegertracing/Lobby',
        },
        {
          label: 'Blog',
          url: 'https://medium.com/jaegertracing',
        },
      ],
    },
  ],
  traceNameRegExpAllowList: FALLBACK_TRACE_NAME_REGEXP_ALLOWLIST,
};