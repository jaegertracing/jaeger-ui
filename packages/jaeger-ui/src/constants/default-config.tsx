// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import deepFreeze from 'deep-freeze';

import { FALLBACK_DAG_MAX_NUM_SERVICES } from './index';

export default deepFreeze(
  Object.defineProperty(
    {
      archiveEnabled: false,
      dependencies: {
        dagMaxNumServices: FALLBACK_DAG_MAX_NUM_SERVICES,
        menuEnabled: true,
      },
      linkPatterns: [],
      menu: [
        {
          label: 'About Jaeger',
          items: [
            {
              label: 'GitHub',
              url: 'https://github.com/uber/jaeger',
            },
            {
              label: 'Docs',
              url: 'http://jaeger.readthedocs.io/en/latest/',
            },
            {
              label: 'Twitter',
              url: 'https://twitter.com/JaegerTracing',
            },
            {
              label: 'Discussion Group',
              url: 'https://groups.google.com/forum/#!forum/jaeger-tracing',
            },
            {
              label: 'Gitter.im',
              url: 'https://gitter.im/jaegertracing/Lobby',
            },
            {
              label: 'Blog',
              url: 'https://medium.com/jaegertracing/',
            },
          ],
        },
      ],
      search: {
        maxLookback: {
          label: '2 Days',
          value: '2d',
        },
        maxLimit: 1500,
      },
      tracking: {
        gaID: null,
        trackErrors: true,
      },
    },
    // fields that should be individually merged vs wholesale replaced
    '__mergeFields',
    { value: ['dependencies', 'search', 'tracking'] }
  )
);

export const deprecations = [
  {
    formerKey: 'dependenciesMenuEnabled',
    currentKey: 'dependencies.menuEnabled',
  },
  {
    formerKey: 'gaTrackingID',
    currentKey: 'tracking.gaID',
  },
];
