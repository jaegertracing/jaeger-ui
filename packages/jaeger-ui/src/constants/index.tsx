// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

export const TOP_NAV_HEIGHT = 47 as 47;

export const FALLBACK_DAG_MAX_NUM_SERVICES = 100 as 100;
export const FALLBACK_TRACE_NAME = '<trace-without-root-span>' as '<trace-without-root-span>';
export const FALLBACK_STANDARD_DATETIME_FORMAT = 'LLL';

export const FETCH_DONE = 'FETCH_DONE' as 'FETCH_DONE';
export const FETCH_ERROR = 'FETCH_ERROR' as 'FETCH_ERROR';
export const FETCH_LOADING = 'FETCH_LOADING' as 'FETCH_LOADING';

export const fetchedState = {
  DONE: FETCH_DONE,
  ERROR: FETCH_ERROR,
  LOADING: FETCH_LOADING,
};
