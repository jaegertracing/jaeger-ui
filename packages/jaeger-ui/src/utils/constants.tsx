// Copyright (c) 2023 The Jaeger Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

/**
 * Provides access to constants injected by the build system.
 */

/**
 * Get the current execution environment, as inferred from NODE_ENV at build time.
 */
export function getAppEnvironment() {
  return __APP_ENVIRONMENT__;
}

/**
 * Get injected version details as a JSON-formatted string.
 */
export function getVersionInfo() {
  return __REACT_APP_VSN_STATE__;
}

export function shouldDebugGoogleAnalytics() {
  return __REACT_APP_GA_DEBUG__;
}
