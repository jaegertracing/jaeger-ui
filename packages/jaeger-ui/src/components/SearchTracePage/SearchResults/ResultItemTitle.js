// @flow

// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';
import { Checkbox } from 'antd';
import { Link } from 'react-router-dom';

import TraceName from '../../common/TraceName';
import { fetchedState } from '../../../constants';
import { formatDuration } from '../../../utils/date';

import type { FetchedState } from '../../../types';
import type { ApiError } from '../../../types/api-error';

import './ResultItemTitle.css';

type Props = {
  duration: number,
  durationPercent: number,
  error?: ApiError,
  isInDiffCohort: boolean,
  linkTo: ?string,
  state: ?FetchedState,
  toggleComparison: (string, boolean) => void,
  traceID: string,
  traceName: string,
  disableComparision?: boolean,
};

export default class ResultItemTitle extends React.PureComponent<Props> {
  props: Props;

  static defaultProps = {
    durationPercent: 0,
    error: undefined,
    state: fetchedState.DONE,
    linkTo: null,
  };

  toggleComparison = () => {
    const { isInDiffCohort, toggleComparison, traceID } = this.props;
    toggleComparison(traceID, isInDiffCohort);
  };

  render() {
    const {
      duration,
      durationPercent,
      error,
      isInDiffCohort,
      linkTo,
      state,
      traceID,
      traceName,
      disableComparision,
    } = this.props;
    // Use a div when the ResultItemTitle doesn't link to anything
    let WrapperComponent = 'div';
    const wrapperProps: { [string]: string } = { className: 'ResultItemTitle--item ub-flex-auto' };
    if (linkTo) {
      WrapperComponent = Link;
      wrapperProps.to = linkTo;
    }
    const isErred = state === fetchedState.ERROR;
    return (
      <div className="ResultItemTitle">
        {!disableComparision && (
          <Checkbox
            className="ResultItemTitle--item ub-flex-none"
            checked={!isErred && isInDiffCohort}
            disabled={isErred}
            onChange={this.toggleComparison}
          />
        )}
        <WrapperComponent {...wrapperProps}>
          <span className="ResultItemTitle--durationBar" style={{ width: `${durationPercent}%` }} />
          {duration != null && <span className="ub-right ub-relative">{formatDuration(duration)}</span>}
          <h3 className="ResultItemTitle--title">
            <TraceName error={error} state={state} traceName={traceName} />
            <small className="ResultItemTitle--idExcerpt">{traceID.slice(0, 7)}</small>
          </h3>
        </WrapperComponent>
      </div>
    );
  }
}
