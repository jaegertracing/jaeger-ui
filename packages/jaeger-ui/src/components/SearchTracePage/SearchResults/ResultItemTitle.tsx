// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';
import { Checkbox } from 'antd';
import { LocationDescriptor } from 'history';
import { Link } from 'react-router-dom';

import TraceName from '../../common/TraceName';
import { fetchedState } from '../../../constants';
import { formatDuration } from '../../../utils/date';

import { FetchedState, TNil } from '../../../types';
import { ApiError } from '../../../types/api-error';

import './ResultItemTitle.css';
import { getTargetEmptyOrBlank } from '../../../utils/config/get-target';

type Props = {
  duration?: number;
  durationPercent?: number;
  error?: ApiError;
  isInDiffCohort: boolean;
  linkTo: LocationDescriptor | TNil;
  state?: FetchedState | TNil;
  targetBlank?: boolean;
  toggleComparison: (traceID: string, isInDiffCohort: boolean) => void;
  traceID: string;
  traceName?: string;
  disableComparision?: boolean;
};

const DEFAULT_DURATION_PERCENT = 0;

const stopCheckboxPropagation = (evt: React.MouseEvent) => evt.stopPropagation();

export default class ResultItemTitle extends React.PureComponent<Props> {
  static defaultProps: Partial<Props> = {
    disableComparision: false,
    durationPercent: DEFAULT_DURATION_PERCENT,
    error: undefined,
    state: undefined,
    targetBlank: false,
  };

  toggleComparison = () => {
    const { isInDiffCohort, toggleComparison, traceID } = this.props;
    toggleComparison(traceID, isInDiffCohort);
  };

  render() {
    const {
      disableComparision,
      duration,
      durationPercent,
      error,
      isInDiffCohort,
      linkTo,
      state,
      targetBlank,
      traceID,
      traceName,
    } = this.props;
    // Use a div when the ResultItemTitle doesn't link to anything
    let WrapperComponent: string | typeof Link = 'div';
    const wrapperProps: Record<string, string | LocationDescriptor> = {
      className: 'ResultItemTitle--item ub-flex-auto',
    };
    if (linkTo) {
      wrapperProps.to = linkTo;
      WrapperComponent = Link;
      if (targetBlank) {
        wrapperProps.target = getTargetEmptyOrBlank();
        wrapperProps.rel = 'noopener noreferrer';
      }
    }
    const isErred = state === fetchedState.ERROR;
    // Separate propagation management and toggle manegement due to ant-design#16400
    const checkboxProps = {
      className: 'ResultItemTitle--item ub-flex-none',
      checked: !isErred && isInDiffCohort,
      disabled: isErred,
      onChange: this.toggleComparison,
      onClick: stopCheckboxPropagation,
    };

    return (
      <div className="ResultItemTitle">
        {!disableComparision && <Checkbox {...checkboxProps} />}
        {/* TODO: Shouldn't need cast */}
        <WrapperComponent {...(wrapperProps as { [key: string]: string; to: string })}>
          <span
            className="ResultItemTitle--durationBar"
            style={{ width: `${durationPercent || DEFAULT_DURATION_PERCENT}%` }}
          />
          {duration != null && <span className="ub-right ub-relative">{formatDuration(duration)}</span>}
          <h3 className="ResultItemTitle--title">
            <TraceName error={error} state={state} traceName={traceName} />
            <small className="ResultItemTitle--idExcerpt">{traceID.slice(0, 7)}</small>
          </h3>
        </WrapperComponent>
      </div>
    );
  }
}
