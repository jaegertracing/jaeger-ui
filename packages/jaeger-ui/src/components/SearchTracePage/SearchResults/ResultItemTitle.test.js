// Copyright (c) 2019 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react';
import { Checkbox } from 'antd';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';

import ResultItemTitle from './ResultItemTitle';
import { fetchedState } from '../../../constants';

describe('ResultItemTitle', () => {
  const defaultProps = {
    duration: 150,
    durationPercent: 10,
    isInDiffCohort: true,
    linkTo: 'linkToValue',
    state: fetchedState.DONE,
    toggleComparison: jest.fn(),
    traceID: 'trace-id-longer-than-8',
    traceName: 'traceNameValue',
  };
  let wrapper;

  beforeEach(() => {
    defaultProps.toggleComparison.mockReset();
    wrapper = shallow(<ResultItemTitle {...defaultProps} />);
  });

  it('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('Checkbox', () => {
    it('does not render toggleComparison checkbox when props.disableComparision is true', () => {
      expect(wrapper.find(Checkbox).length).toBe(1);
      wrapper.setProps({ disableComparision: true });
      expect(wrapper.find(Checkbox).length).toBe(0);
    });

    it('is disabled iff props.state === fetchedState.ERROR', () => {
      expect(wrapper.find(Checkbox).prop('disabled')).toBe(false);
      wrapper.setProps({ state: fetchedState.ERROR });
      expect(wrapper.find(Checkbox).prop('disabled')).toBe(true);
    });

    it('is checked iff props.state !== fetchedState.ERROR && props.isInDiffCohort', () => {
      [true, false].forEach(isInDiffCohort => {
        [fetchedState.ERROR, fetchedState.DONE].forEach(state => {
          wrapper.setProps({ isInDiffCohort, state });
          expect(wrapper.find(Checkbox).prop('checked')).toBe(state !== fetchedState.ERROR && isInDiffCohort);
        });
      });
    });

    it('calls props.toggleComparison with correct arguments onChange', () => {
      wrapper.find(Checkbox).prop('onChange')();
      expect(defaultProps.toggleComparison).toHaveBeenCalledWith(
        defaultProps.traceID,
        defaultProps.isInDiffCohort
      );
      wrapper.setProps({ isInDiffCohort: !defaultProps.isInDiffCohort });
      wrapper.find(Checkbox).prop('onChange')();
      expect(defaultProps.toggleComparison).toHaveBeenLastCalledWith(
        defaultProps.traceID,
        !defaultProps.isInDiffCohort
      );
    });
  });

  describe('WrapperComponent', () => {
    it('renders <Link> when linkTo is provided', () => {
      expect(wrapper.find(Link).length).toBe(1);
      wrapper.setProps({ linkTo: null });
      expect(wrapper.find(Link).length).toBe(0);
    });

    it('<Link> targets _blank and sets rel when targetBlank is true', () => {
      expect(wrapper.find(Link).prop('target')).toBeUndefined();
      expect(wrapper.find(Link).prop('rel')).toBeUndefined();
      wrapper.setProps({ targetBlank: true });
      expect(wrapper.find(Link).prop('target')).toBe('_blank');
      expect(wrapper.find(Link).prop('rel')).toBe('noopener noreferrer');
    });

    it('hides formated duration when duration is not provided', () => {
      const initialSpanCount = wrapper.find('span').length;
      wrapper.setProps({ duration: null });
      expect(wrapper.find('span').length).toBe(initialSpanCount - 1);
    });
  });
});
