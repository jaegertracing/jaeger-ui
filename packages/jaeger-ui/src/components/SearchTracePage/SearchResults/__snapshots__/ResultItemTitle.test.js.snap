// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItemTitle renders as expected 1`] = `
<div
  className="ResultItemTitle"
>
  <Checkbox
    checked={true}
    className="ResultItemTitle--item ub-flex-none"
    disabled={false}
    indeterminate={false}
    onChange={[Function]}
    prefixCls="ant-checkbox"
  />
  <Link
    className="ResultItemTitle--item ub-flex-auto"
    replace={false}
    to="linkToValue"
  >
    <span
      className="ResultItemTitle--durationBar"
      style={
        Object {
          "width": "10%",
        }
      }
    />
    <span
      className="ub-right ub-relative"
    >
      0.15ms
    </span>
    <h3
      className="ResultItemTitle--title"
    >
      <TraceName
        className=""
        error={null}
        state="FETCH_DONE"
        traceName="traceNameValue"
      />
      <small
        className="ResultItemTitle--idExcerpt"
      >
        trace-i
      </small>
    </h3>
  </Link>
</div>
`;
